chain/vm/vmi.go

```diff
@@ -27,13 +37,24 @@ type Interface interface {
 	Flush(ctx context.Context) (cid.Cid, error)
 }
 
+// Executor is the general vm execution interface, which is prioritized according to execution lanes.
+// User must call Done when it is done with this executor to release resource holds by the execution
+// environment
+type Executor interface {
+	Interface
+
+	// Done must be called when done with the executor to release resource holds.
+	// It is an error to invoke Interface methods after Done has been called.
+	Done()
```

Comment:
So, I share this concern -- honestly, I think the right thing to do here is to just attach a lifespan to the executor / executionToken. There's basically no case in which we want a single executor to be lasting for, say, 15 minutes. If anything has taken that long, something has gone wrong, just shut it down.

Unfortunately, I think this is hard to do safely, because you really don't want to be aborting execution midway...

Reply:
Yeah, this is hard to do in go due to the lack of finalizers.
+1 for the crab here, you'd just implement drop...

Reply:
resolved -- Done no more.